fud2/fud-core/src/run.rs

```diff
@@ -32,6 +35,13 @@ impl std::fmt::Display for RunError {
             RunError::MissingConfig(s) => {
                 write!(f, "missing required config key: {}", s)
             }
+            RunError::InvalidValue(key, value, values) => {
+                write!(
+                    f,
+                    "invalid value '{}' for key '{}'. Valid values are {:?}",
```

Comment:
Instead of `{:?}`, which will generate the debug representation for the string, can you do: `values.into_iter().join(", ")` instead?

Reply:
Sure, out of curiosity why do we prefer joins over the debug representation?

Reply:
FWIW I concur here; it just makes for a slightly nicer UI. Compare:

```
valid values are ["foo", "bar", "baz"]
valid values are foo, bar, baz
```

The latter is a bit more like how humans communicate. :smiley: